File: Problems 1-9/problem2.c
#include <stdio.h>
 
int main()
{
	unsigned int a = 1, b = 1, c = 2, sum = 0;
	
	while (c < 4000000)
	{
		c = a + b;
		sum += c * !(c % 2);
		a = b;
		b = c;
	}
	
	printf("%u\n", sum);
	
	return 0;
}

File: Problems 1-9/problem2.py
def fib(limit = 4000000):
	"""Returns the cumulative sum of all the even Fibonacci numbers up to a given limit"""
	a, b, s = 1, 0, 0
	while s < limit:
		a, b = a+b, a
		if a % 2 == 0:
			s += a
	
	return s